pgjdbc/src/main/java/org/postgresql/jdbc/PgResultSet.java

```diff
@@ -2098,9 +2098,6 @@ public long getLong(int columnIndex) throws SQLException {
     if (isBinary(columnIndex)) {
       int col = columnIndex - 1;
       int oid = fields[col].getOID();
-      if (oid == Oid.INT8) {
-        return ByteConverter.int8(this_row[col], 0);
```

Comment:
Of course it doesn't fail.
In order to make it fail, you need:
1) invoke the `insert` via regular non-batch API multiple times
2) then invoke the same `insert` via batch API

1 and 2 should be in the same test method (since JUnit's before/after table drop/create are performed in between test method execution)